receiver/prometheusreceiver/internal/otlp_metricfamily.go

```diff
@@ -170,10 +178,25 @@ func (mg *metricGroup) toDistributionPoint(orderedLabelKeys []string, dest *pmet
 	}
 	point.SetTimestamp(tsNanos)
 	populateAttributes(orderedLabelKeys, mg.ls, point.Attributes())
-
+	mg.setExemplars(exemplarsdata, point.Exemplars())
 	return true
 }
 
+func (mg *metricGroup) setExemplars(exemplarMap map[string][]ExemplarData, exemplars pmetric.ExemplarSlice) {
+	metricKey := buildMetricKey(mg.ls)
+	ed := exemplarMap[metricKey]
+	es := pmetric.NewExemplarSlice()
```

Comment:
Why create a new slice here and copy it to the incoming slice later?  Why not simply ensure capacity and append empty exemplars to the incoming slice?  Or create and return a new slice?